java/client/src/main/java/glide/api/models/commands/SetOptions.java

```diff
@@ -0,0 +1,129 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import glide.api.commands.StringCommands;
+import glide.api.models.exceptions.RequestException;
+import java.util.LinkedList;
+import java.util.List;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import redis_request.RedisRequestOuterClass.Command;
+
+/**
+ * Optional arguments for {@link StringCommands#set(String, String, SetOptions)} command.
+ *
+ * @see <a href="https://redis.io/commands/set/">redis.io</a>
+ */
+@Builder
+public final class SetOptions {
+
+    /**
+     * If <code>conditionalSet</code> is not set the value will be set regardless of prior value
+     * existence. If value isn't set because of the condition, command will return <code>null</code>.
+     */
+    private final ConditionalSet conditionalSet;
+
+    /**
+     * Set command to return the old string stored at <code>key</code>, or <code>null</code> if <code>
+     * key</code> did not exist. An error is returned and <code>SET</code> aborted if the value stored
+     * at <code>key</code> is not a string. Equivalent to <code>GET</code> in the Redis API.
+     */
+    private final boolean returnOldValue;
+
+    /** If not set, no expiry time will be set for the value. */
+    private final TimeToLive expiry;
+
+    /** Conditions which define whether new value should be set or not. */
+    @RequiredArgsConstructor
+    @Getter
+    public enum ConditionalSet {
+        /**
+         * Only set the key if it does not already exist. Equivalent to <code>XX</code> in the Redis
+         * API.
+         */
+        ONLY_IF_EXISTS("XX"),
+        /** Only set the key if it already exists. Equivalent to <code>NX</code> in the Redis API. */
+        ONLY_IF_DOES_NOT_EXIST("NX");
+
+        private final String redisApi;
+    }
+
+    /** Configuration of value lifetime. */
+    @Builder
+    public static final class TimeToLive {
+        /** Expiry type for the time to live */
+        @NonNull private TimeToLiveType type;
+
+        /**
+         * The amount of time to live before the key expires. Ignored when {@link
+         * TimeToLiveType#KEEP_EXISTING} type is set.
+         */
+        private Integer count;
```

Comment:
Any reason this needs to be an Integer over an int?

Reply:
int cannot be defaulted to null